File: ct-incremental-checklist/run-checklist.mjs
#!/usr/bin/env node
import { runChecklist } from "./checks.mjs";

const version            = process.env.CT_VERSION;
const date_release_build = process.env.CT_BUILD_DATE;

if (!version || !date_release_build) {
  console.error("Error: CT_VERSION and CT_BUILD_DATE environment variables must be set.");
  console.error("  e.g. CT_VERSION=2.1.13 CT_BUILD_DATE=20260303 node run-checklist.mjs");
  process.exit(2);
}

const { lines, allOk } = await runChecklist(version, date_release_build);
console.log(lines.join("\n"));
process.exit(allOk ? 0 : 1);

File: ct-incremental-checklist/run_checklist.py
#!/usr/bin/env python3
import asyncio
import os
import sys

from checks import run_checklist


async def main():
    version = os.environ.get("CT_VERSION")
    date_release_build = os.environ.get("CT_BUILD_DATE")

    if not version or not date_release_build:
        print("Error: CT_VERSION and CT_BUILD_DATE environment variables must be set.", file=sys.stderr)
        print("  e.g. CT_VERSION=2.1.13 CT_BUILD_DATE=20260303 python run_checklist.py", file=sys.stderr)
        sys.exit(2)

    lines, all_ok = await run_checklist(version, date_release_build)
    print("\n".join(lines))
    sys.exit(0 if all_ok else 1)


if __name__ == "__main__":
    asyncio.run(main())
